### PR TITLE
Add change password workflow

### PR DESCRIPTION
**Pull request by Rubikium**
ref #2895

### CODE REVIEW

File: pkg/latte/intent_change_password.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+func (*IntentChangePassword) JSONSchema() *validation.SimpleSchema {
+	return IntentChangePasswordSchema
+}
+
+func (*IntentChangePassword) CanReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow) ([]workflow.Input, error) {
+	if len(w.Nodes) == 0 {
+		return []workflow.Input{
+			&InputTakeLoginID{},
```
Comment (carmenlau) [inline pkg/latte/intent_change_password.go]: Changing the password is a feature that requires an authenticated user, `IntentChangePassword` should not take the login id to search the user.

Instead, the user id should be resolved from the user session which can be obtained through the context. e.g. https://github.com/authgear/authgear-server/blob/c81d8eda8da7ceba70ccc853f25a3c692957abe3/pkg/latte/intent_verify_user.go#L44

File: pkg/latte/intent_change_password.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/authgear/authgear-server/pkg/lib/authn/identity"
+	"github.com/authgear/authgear-server/pkg/lib/workflow"
+	"github.com/authgear/authgear-server/pkg/util/validation"
+)
+
+func init() {
+	workflow.RegisterPublicIntent(&IntentChangePassword{})
+}
+
+var IntentChangePasswordSchema = validation.NewSimpleSchema(`{}`)
```
Comment (carmenlau) [inline pkg/latte/intent_change_password.go]: Public intent schema cannot be empty, see `IntentVerifyUser`.

File: pkg/latte/input_take_change_password.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		"additionalProperties": false,
+		"properties": {
+			"old_password": { "type": "string" },
+			"new_password": { "type": "string" }
+		},
+		"required": ["old_password", "new_password"]
+	}
+`)
+
+type InputTakeChangePassword struct {
```
Comment (carmenlau) [inline pkg/latte/input_take_change_password.go:24]: Maybe we can call the input `InputChangePassword`? `InputTakeChangePassword` has 2 verbs.

File: pkg/latte/node_change_password.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+func (n *NodeChangePassword) ReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow, input workflow.Input) (*workflow.Node, error) {
+	var inputTakeChangePassword inputTakeChangePassword
+	switch {
+	case workflow.AsInput(input, &inputTakeChangePassword):
+		info, err := n.getPasswordAuthenticator(deps)
+		// The user doesn't have the password authenticator
+		// always returns invalid credentials error
+		if errors.Is(err, api.ErrNoAuthenticator) {
+			return nil, api.ErrInvalidCredentials
```
Comment (louischan-oursky) [inline pkg/latte/node_change_password.go:44]: 1. Move `ChangePasswordFailed` and `ErrNoPassword` in pkg/lib/interaction/nodes/change_password.go to pkg/api/errors.go
2. Return `ErrNoPassword` when the user has no password

File: pkg/latte/node_change_password_end.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	"github.com/authgear/authgear-server/pkg/lib/authn/authenticator"
+	"github.com/authgear/authgear-server/pkg/lib/workflow"
+)
+
+func init() {
+	workflow.RegisterNode(&NodeChangePasswordEnd{})
+}
+
+type NodeChangePasswordEnd struct {
+	Authenticator *authenticator.Info
```
Comment (louischan-oursky) [inline pkg/latte/node_change_password_end.go:15]: Remove this field.

**Pull request merged**